### PR TITLE
ci: improve PR template with additional context

### PR DESCRIPTION
**Pull request by miguelhrocha**
Hello!

I wanted to open a PR with a PR template that will allow us to remember to add better context/descriptions in future PRs. Let me know whether you agree with these sections, or if you'd like to see any additional ones!

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+- [ ] `pip install -e .`, start server and executor, cd to `python-sdk/tests`, `python test_graph_behaviours.py`.
+
+## How to test
+
+<!-- Please describe in detail how to test the changes. -->
+
+## Screenshots
+
+<!-- If applicable, add screenshots to help explain your problem. -->
+
```
Comment (seriousben) [inline .github/pull_request_template.md:25]: I think this is a great idea!

Proposing another version of the template for your consideration.
Your version is great, I just wanted to provide an alternative to see if we could get to a better version together. ✋ 5️⃣ 

```suggestion
## Context

<!--
In a few sentences or less, please explain the context behind this change to help answer why this change is needed.

If this is a bug fix, make sure to include "fixes #xxxx", or
"closes #xxxx".

Screenshots, logs, code or other visual aids are greatly appreciated.
 -->

## What

<!--
In a few sentences, please summarize the change to help reviewers.

Consider providing screenshots, logs, code or other visual aids to help the reviewer understand the approach taken.
-->

## Testing

<!--
Please include steps used to verify the change.

Consider providing screenshots, logs, code or other visual aids to help the reviewer in their testing.
-->

## Contribution Checklist

- [ ] If the python-sdk was changed, please run `make fmt` in `python-sdk/`.
- [ ] If the server was changed, please run `make fmt` in `server/`.
- [ ] Make sure all PR Checks are passing.
<!-- You can run the tests manually:
In `python-sdl/`, run the run `pip install -e .`, start the server and executor, `python test_graph_behaviours.py`.

```

**Review by seriousben — APPROVED** [review]
Amazing!
I provided a possible alternative which I feel might be simpler.

But I have no strong opinions.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+Consider providing screenshots, logs, code or other visual aids to help the reviewer in their testing.
+-->
+
+## Contribution Checklist
+
+- [ ] If the python-sdk was changed, please run `make fmt` in `python-sdk/`.
+- [ ] If the server was changed, please run `make fmt` in `server/`.
+- [ ] Make sure all PR Checks are passing.
+<!-- You can run the tests manually:
+In `python-sdl/`, run the run `pip install -e .`, start the server and executor, `python test_graph_behaviours.py`.
```
Comment (seriousben) [inline .github/pull_request_template.md:34]: My typo :) 

```suggestion
In `python-sdk/`, run the run `pip install -e .`, start the server and executor, `python test_graph_behaviours.py`.
```

**Comment by miguelhrocha** [comment]
> For OSS Contributors, do we make them sign a CLA before the contributions are accepted? More Info: https://cla-assistant.io/

That is a good idea, but I believe it must be done as a GHA instead of appending it to a PR template 🤔

**Comment by diptanu** [comment]
Yeah we need a CLA, I am working with our lawyers to figure this out.

**Pull request merged**